### PR TITLE
Improved queue, test with Boost Test, updated driver

### PR DESCRIPTION
**Pull request by Lorak-mmk**
- Updated scmd version: now we can use async queries and other cool things.
- Refactored CMake a bit
- Tests now use boost test
- There is now one executable (both for workers and init), which has basic command line options parsing.

New queue implementation:
- Definitely more flexible and useful interface - you can produce/consume tasks in any order.
- Supports any combination of single/multi producer and single/multi consumer, as you can see in tests
- Uses optimistic locking technique, so it would be best to avoid heavy use, as it may negatively affect performance. It would be best to split tasks into as large parts as possible to avoid such problems. Still, it may be a bit faster than previous implementation as it doesn't DELETE old tasks immediately (so no tombstones) and doesn't use counters - but that is pure speculation, I didn't run any benchmarks.

I'd love some feedback ^_^ Also, does anyone know how to properly test such things as the queue? I wrote tests using Boost Test, but they are single-threaded - we produce items, and then consume them. It would be nice to test it for race conditions, have multiple consumers etc, but I'm not sure how to approach it - I'm open to any ideas.

### CODE REVIEW

**Review by jbhayven — APPROVED** [review]
Looks ok, but a brief comment-based documentation of queue methods in its header file would be much appreciated.

**Review by jbhayven — APPROVED** [review]

File: src/scylla_blas/queue/scylla_queue.cc [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+[[maybe_unused]] void scylla_blas::scylla_queue::deinit_meta(const std::shared_ptr<scmd::session> &session) {
+    session->execute("DROP TABLE IF EXISTS blas.queue_meta");
+}
+
+bool scylla_blas::scylla_queue::queue_exists(const std::shared_ptr<scmd::session> &session, int64_t id) {
+    auto result = session->execute(fmt::format("SELECT * FROM blas.queue_meta WHERE id = {}", id));
+    return result.row_count() == 1;
+}
+
+void
```
Comment (Rhantolq) [inline src/scylla_blas/queue/scylla_queue.cc:23]: void\n?

**Review by Rhantolq — CHANGES_REQUESTED** [review]
A file was moved to tests/generators/value_factory.hh
We will probably let users use our generators, so that they can supply us with their own values for matrices, so we probably want to leave some abstract class in sources.

File: src/main.cc [hunk 2]
```diff
@@ -34,7 +34,7 @@ void parse_arguments(int ac, char *av[], options &options) {
             ("init", "Initialize Scylla keyspace and tables")
             ("worker", "Connect to Scylla and process incoming requests")
             ("host,H", po::value<std::string>(&options.host)->required(), "Address on which Scylla can be reached")
-            ("port,P", po::value<uint16_t>(&options.port)->required(), "port number on which Scylla can be reached");
+            ("port,P", po::value<uint16_t>(&options.port)->default_value(9042), "port number on which Scylla can be reached");
```
Comment (jbhayven) [inline src/main.cc:37]: Move the magic number elsewhere? A new config.cc, perhaps?

**Review by jbhayven — COMMENTED** [review]
How about defaulting the host address in tests to the same value as that in default-run run_scylla.sh, i.e. `172.19.0.2`?

**Review by Rhantolq — APPROVED** [review]

**Review by jbhayven — APPROVED** [review]

**Review by jbhayven — APPROVED** [review]

**Comment by Lorak-mmk** [comment]
- Updated scmd version: now we can use async queries and other cool things.
 - Refactored CMake a bit
 - Tests now use boost test
 - There is now one executable (both for workers and init), which has basic command line options parsing.
 - New implementation of queue

**Pull request merged**